lib/flight_scheduler/allocation_registry.rb

```diff
@@ -93,6 +93,17 @@ def add(allocation)
       # Duplicate the allocation so it is safe to modify
       allocation = allocation.dup
 
+      # NOTE: Updating the state of the internal job object violates the Law of Demeter
+      #       as the registry needs to "reach through" the allocation first.
+      #
+      #       Whilst not ideal, it is important the AllocationRegistry and CONFIGURING
+      #       state stay in sync with each other.
+      #
+      #       Consider refactoring to do the following:
+      #       1. Infer the Job's state directly from the registry, or
+      #       2. Change add(allocation) to build(job, nodes). This would likely require
+      #          the use of 'Reservations' instead of 'Allocations' in the scheduler
+      allocation.job.state = 'CONFIGURING' if allocation.job.pending?
```

Comment:
What do you mean by inferring the job's state from the registry?  Something like this?

```
def state
  return super unless super == 'PENDING'
  allocated? ? 'CONFIGURING' : 'PENDING'
end
```

Reply:
Something along those lines would work, however I was hesitant to break the established design pattern around the `state`.